testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java

```diff
@@ -112,9 +112,12 @@ protected void doTestHiveCoercion(HiveTableDefinition tableDefinition)
 
         Function<Engine, Map<String, List<Object>>> expected = engine -> expectedValuesForEngineProvider(engine, tableName, decimalToFloatVal, floatToDecimalVal);
 
+        // For Trino, remove unsupported columns
         List<String> prestoReadColumns = removeUnsupportedColumnsForTrino(allColumns, tableName);
         Map<String, List<Object>> expectedPrestoResults = expected.apply(Engine.TRINO);
-        assertEquals(ImmutableSet.copyOf(prestoReadColumns), expectedPrestoResults.keySet());
+        if (expectedExceptionsWithTrinoContext().isEmpty()) {
+            assertEquals(ImmutableSet.copyOf(prestoReadColumns), expectedPrestoResults.keySet());
```

Comment:
In case of unpartitioned tables we might not support all the column coercion as of now so we avoid the assertion when Trino throws some sort of exception.

Reply:
add a code comment then